File: src/runtime/boxes/succeed/bb_succeed.cs
// bb_succeed.cs — SUCCEED: always γ zero-width; outer scan loop retries
// Mirrors src/runtime/boxes/succeed/bb_succeed.c
//
// Always succeeds; the scan loop's retry-on-backtrack produces the
// "try-everything" effect SNOBOL4's SUCCEED provides.

namespace Snobol4.Runtime.Boxes;

public sealed class bb_succeed : IByrdBox
{
    public Spec α(MatchState ms) => Spec.ZeroWidth(ms.Cursor);
    public Spec β(MatchState ms)  => Spec.ZeroWidth(ms.Cursor);
}

File: src/runtime/boxes/succeed/bb_succeed.java
package bb;

/**
 * bb_succeed.java — SUCCEED: always γ zero-width; outer scan loop retries
 * Port of bb_succeed.c / bb_succeed.s
 */
public class bb_succeed extends bb_box {

    public bb_succeed(MatchState ms) { super(ms); }

    @Override public Spec alpha() { return new Spec(ms.delta, 0); }
    @Override public Spec beta()  { return new Spec(ms.delta, 0); }
}
